ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -605,6 +613,7 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
                     StoreErrorCodes.Life_Version_Conflict);
               }
             }
+            indexValuesToDelete.set(i, value);
```

Comment:
minor: can we call this `indexValuePriorToNewDelete` or the more descriptive name?